server/ice_servers/coturn.py

```diff
@@ -0,0 +1,45 @@
+"""
+Coturn static auth api
+
+Creates ice_server record with temporary credentials computed from coturn static-auth-secret
+
+http://stackoverflow.com/a/35767224
+"""
+
+import time
+import hmac
+import base64
+from hashlib import sha1
+from server.config import COTURN_HOSTS, COTURN_KEYS, TWILIO_TTL
+
+class CoturnHMAC:
+    def __init__(self, coturn_hosts=None, coturn_keys=None):
+        self.coturn_hosts = coturn_hosts or COTURN_HOSTS
+        self.coturn_keys = coturn_keys or COTURN_KEYS
+
+    def fetch_token(self, username='faf-user', ttl=None):
+        servers = []
+
+        ttl = ttl or TWILIO_TTL
+        if ttl is not None:
+            ttl = int(ttl)
+        else:
+            ttl = 3600*24
+
+        # See https://github.com/coturn/coturn/wiki/turnserver#turn-rest-api
+        # create hmac of coturn_key + timestamp:username
```

Comment:
This comment tells me that the following code should be extracted to a function for readability. You could probably make it a static method

Reply:
it is already extracted to a function, everything this function does is generate the long term credential

Reply:
So, this comment is purely a "Clean code" nit pick. Although reading this code more closely I think the function name is misleading.

This function does about 5 things:
1. A bunch of weird (IMO unnecessary) logic for determining what the ttl should be
2. Iterates through all of the coturn hosts:
    1. Computes the hmac for each host
    2. Computes the host urls
3. Aggregates all of the computed keys and hosts to a single list and returns that list

Ideally the contents of the loop would go into their own function which actually calculates ONE token. This function is called `fetch_token` which implies that it will return ONE token. But it doesn't, it returns a list of dictionaries which contain a bunch of information, among other things, several tokens.

Reply:
yeah, we could just extract the generation of a single credential, and I agree, that function name is chosen very badly